xscen/extract.py

```diff
@@ -1110,65 +1124,158 @@ def subset_warming_level(
        The dataset should include attributes to help recognize it and find its
        warming levels - 'cat:mip_era', 'cat:experiment', 'cat:member', and either
        'cat:source' for global models or 'cat:driving_institution' (optional) + 'cat:driving_model' for regional models.
-    wl : float
+       Or , it should include a `realization` dimension constructed as "{mip_era}_{source or driving_model}_{experiment}_{member}"
+       for vectorized subsetting. Vectorized subsetting is currently only implemented for annual data.
+    wl : float or sequence of floats
        Warming level.
        eg. 2 for a global warming level of +2 degree Celsius above the mean temperature of the `tas_baseline_period`.
+       Multiple levels can be passed, in which case using "{wl}" in  `to_level` and `wl_dim` is not recommended.
+       Mutliple levels are currently only implemented for annual data.
     to_level :
        The processing level to assign to the output.
        Use "{wl}", "{period0}" and "{period1}" in the string to dynamically include
        `wl`, 'tas_baseline_period[0]' and 'tas_baseline_period[1]'.
-    wl_dim : str
+    wl_dim : str or boolean, optional
        The value to use to fill the new `warminglevel` dimension.
        Use "{wl}", "{period0}" and "{period1}" in the string to dynamically include
        `wl`, 'tas_baseline_period[0]' and 'tas_baseline_period[1]'.
-       If None, no new dimensions will be added.
-
-    kwargs
-        Instructions on how to search for warming levels.
-        The keyword arguments are passed to `get_warming_level()`
-
-        Valid keyword aguments are:
-            window : int
-            tas_baseline_period : list
-            ignore_member : bool
-            tas_csv : str
-            return_horizon: bool
+       If None, no new dimensions will be added, invalid if `wl` is a sequence.
+       If True, the dimension will include `wl` as numbers and units of "degC".
+    kwargs :
+        Instructions on how to search for warming levels, passed to :py:func:`get_warming_level`.
 
     Returns
     -------
-    xr.Dataset
-        Warming level dataset.
+    xr.Dataset or None
+        Warming level dataset, or None if `ds` can't be subsetted for the requested warming level.
+        The dataset will have a new dimension `warminglevel` with `wl_dim` as coordinates.
```

Comment:
what's the point of keeping the fake time axis ?

Reply:
Maybe I don't understand the question, but as `get_warming_level` does not perform any statistical computation, there's a N year window for each warming level. The only way to have a multidimensional dataset with  `time` and `wl_dim`, is to have some sort of time axis that is not the actual horizon (which is different for each warming level and simulation).

Reply:
my mistake. I forgot that the mean wasn't calculated on the window in that function.